oauthenticator/oauth2.py

```diff
@@ -426,8 +481,274 @@ def get_handlers(self, app):
             (r'/logout', self.logout_handler),
         ]
 
-    async def authenticate(self, handler, data=None):
-        raise NotImplementedError()
+    def build_userdata_request_headers(self, access_token, token_type):
+        """
+        Builds and returns the headers to be used in the userdata request.
+        Called by the :meth:`oauthenticator.OAuthenticator.token_to_user`
+        """
+        return {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "User-Agent": "JupyterHub",
+            "Authorization": "{} {}".format(token_type, access_token),
+        }
+
+    def build_token_info_req_headers(self):
```

Comment:
Consistency with userdata req->request:

```suggestion
    def build_token_info_request_headers(self):
```